tests/e2e/mgmt_wrkld_domain_isolation_negative.go

```diff
@@ -0,0 +1,485 @@
+		defer func() {
+			delTestWcpNs(vcRestSessionId, namespace)
+			gomega.Expect(waitForNamespaceToGetDeleted(ctx, client, namespace, poll, pollTimeout)).To(gomega.Succeed())
+		}()
+
+		ginkgo.By("Read zonal class")
+		storageclassZ1, err := client.StorageV1().StorageClasses().Get(ctx, storagePolicyNameZ1, metav1.GetOptions{})
+		if !apierrors.IsNotFound(err) {
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+		storageclassZ2, err := client.StorageV1().StorageClasses().Get(ctx, storagePolicyNameZ2, metav1.GetOptions{})
+		if !apierrors.IsNotFound(err) {
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		ginkgo.By("Creating pvc")
+		pvclaim1, _, err := createPVCAndQueryVolumeInCNS(ctx, client, namespace, labelsMap, "",
+			diskSize, storageclassZ1, true)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Wait for PVC to reach Bound state.")
+		_, err = fpv.WaitForPVClaimBoundPhase(ctx, client,
+			[]*v1.PersistentVolumeClaim{pvclaim1}, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Create Pod to attach to Pvc-1")
+		pod, err := createPod(ctx, client, namespace, nil, []*v1.PersistentVolumeClaim{pvclaim1}, false,
+			execRWXCommandPod1)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Verify svc pv affinity, pvc annotation and pod node affinity")
+		allowedTopologies = setSpecificAllowedTopology(allowedTopologies, topkeyStartIndex, 0,
+			1)
+		err = verifyPvcAnnotationPvAffinityPodAnnotationInSvc(ctx, client, nil, pod, nil, namespace,
+			allowedTopologies)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Creating service")
+		_ = CreateService(namespace, client)
+
+		ginkgo.By("Creating statefulset")
+		sts := createCustomisedStatefulSets(ctx, client, namespace, true, replicas, false, nil,
+			false, true, "", "", storageclassZ2, storagePolicyNameZ2)
+		allowedTopologies = setSpecificAllowedTopology(allowedTopologies, topkeyStartIndex, 1,
+			2)
+		err = verifyPvcAnnotationPvAffinityPodAnnotationInSvc(ctx, client, sts, nil, nil, namespace,
+			allowedTopologies)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Mark zone-2 for removal from wcp namespace")
+		err = markZoneForRemovalFromWcpNs(vcRestSessionId, namespace,
+			zone2)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		// get ds name
+		pv := getPvFromClaim(client, namespace, pvclaim1.Name)
+		volHandle := pv.Spec.CSI.VolumeHandle
+		s1 := rand.NewSource(time.Now().UnixNano())
+		r1 := rand.New(s1)
+		randomStr := strconv.Itoa(r1.Intn(1000))
+		datastoreName, dsRef, _ := e2eVSphere.fetchDatastoreNameFromDatastoreUrl(ctx, volHandle)
+
+		ginkgo.By("Stop Vsan-health service")
+		ginkgo.By(fmt.Sprintln("Stopping vsan-health on the vCenter host"))
+		isVsanHealthServiceStopped = true
+		err = invokeVCenterServiceControl(ctx, stopOperation, vsanhealthServiceName, vcAddress)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
```

Comment:
In case if any test step fails..it might get skipped so keep it this code in AfterEach block as well